homeassistant/components/aurora_abb_powerone/sensor.py

```diff
@@ -61,70 +70,47 @@ async def async_setup_entry(
 
 
-class AuroraSensor(AuroraEntity, SensorEntity):
+class AuroraSensor(CoordinatorEntity[AuroraAbbDataUpdateCoordinator], SensorEntity):
     """Representation of a Sensor on a Aurora ABB PowerOne Solar inverter."""
 
     _attr_has_entity_name = True
 
     def __init__(
         self,
-        client: AuroraSerialClient,
+        coordinator: AuroraAbbDataUpdateCoordinator,
         data: Mapping[str, Any],
         entity_description: SensorEntityDescription,
     ) -> None:
         """Initialize the sensor."""
-        super().__init__(client, data)
+        super().__init__(coordinator)
+        self._data = data
         self.entity_description = entity_description
-        self.available_prev = True
-
-    def update(self) -> None:
-        """Fetch new state data for the sensor.
-
-        This is the only method that should fetch new data for Home Assistant.
-        """
-        try:
-            self.available_prev = self._attr_available
-            self.client.connect()
-            if self.entity_description.key == "instantaneouspower":
-                # read ADC channel 3 (grid power output)
-                power_watts = self.client.measure(3, True)
-                self._attr_native_value = round(power_watts, 1)
-            elif self.entity_description.key == "temp":
-                temperature_c = self.client.measure(21)
-                self._attr_native_value = round(temperature_c, 1)
-            elif self.entity_description.key == "totalenergy":
-                energy_wh = self.client.cumulated_energy(5)
-                self._attr_native_value = round(energy_wh / 1000, 2)
-            self._attr_available = True
-
-        except AuroraTimeoutError:
-            self._attr_state = None
-            self._attr_native_value = None
-            self._attr_available = False
-            _LOGGER.debug("No response from inverter (could be dark)")
-        except AuroraError as error:
-            self._attr_state = None
-            self._attr_native_value = None
-            self._attr_available = False
-            raise error
-        finally:
-            if self._attr_available != self.available_prev:
-                if self._attr_available:
-                    _LOGGER.info("Communication with %s back online", self.name)
-                else:
-                    _LOGGER.warning(
-                        "Communication with %s lost",
-                        self.name,
-                    )
-            if self.client.serline.isOpen():
-                self.client.close()
+        self._attr_unique_id = (
+            f"{data[ATTR_SERIAL_NUMBER]}_{self.entity_description.key}"
+        )
+
+    @property
+    def native_value(self) -> StateType:
+        """Get the value of the sensor from previously collected data."""
+        return self.coordinator.data.get(self.entity_description.key)
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device specific attributes."""
+        return {
```

Comment:
Why not use the DeviceInfo object and why use the short hand attribute?

Reply:
Agreed, done [7760e68](https://github.com/home-assistant/core/pull/72363/commits/7760e68e93172c5ebd3064a8a10bbd5c8ef5fc76). Thanks!